kernel/src/memory/heap.rs

```diff
@@ -144,15 +168,30 @@ impl Heap {
 
             PAGE_TABLES.lock().unmap(
                 Page::containing_address(page_addr, PageSize::Kib4),
-                false,
+                false, // Do not free backing memory
+                true, // Do invplg
             );
         }
     }
+
+    pub const fn tree_size() -> usize {
+        mem::size_of::<[Block; BLOCKS_IN_TREE]>()
+    }
+
+    pub fn is_free(&self, ptr: *const u8, layout: Layout) {
+        let order = order(layout.size());
+        let global_ptr = ptr;
+        let ptr = ptr as usize - HEAP_START;
+
+        let level = MAX_ORDER - order;
+        let level_offset = super::buddy_allocator::blocks_in_level(level);
+        let index = level_offset + ((ptr as usize) >> (order + 6));
```

Comment:
6 seems pretty arbitrary -- maybe extract to constant?

Reply:
6 is the base order. Will extract though